File: linearsearch.c
#include<stdio.h>
main()
{
int a[10],i,n,s ,f=0;
printf("enter the no.s");
scanf("%d",&n);
for(i=0;i<n;i++)
scanf("%d",&a[i]);
printf("enter the no. to be checked");
scanf("%d",&s);
for(i=0;i<n;i++)
{
if (a[i]==s)
{
f=1;
break;}}
if(f==1)
printf("the %d is present in the position %d",s,i+1);
else 
printf("the no. is not present");
}

File: LinearSearch.java
import java.util.*;
import java.io.*;
class Sub
{int n;
Scanner s=new Scanner(System.in);
int a[]=new int[50];
int i=0;
int flag=0,key,pos;
void input()
{
System.out.println("enter limit");
 n=s.nextInt();
System.out.println("enter elements");
for(i=0;i<n;i++)
{
a[i]=s.nextInt();
}
System.out.println("element to be searched?");
key=s.nextInt();
}
void search()
{
for(i=0;i<n;i++)
{
if(a[i]==key)
{
flag=1;
break;
}}
if(flag==1)
{
System.out.println("element to be searched is found at position"+(i+1));
}
else{
System.out.println("element not found!!!! Search failed");
}}}
class LinearSearch
{
public static void main(String args[])
{
Sub a=new Sub();
a.input();
a.search();
}}






